Source/WebCore/rendering/FloatingObjects.cpp

```diff
@@ -439,6 +439,17 @@ LayoutUnit FloatingObjects::logicalRightOffset(LayoutUnit fixedOffset, LayoutUni
     return std::min(fixedOffset, adapter.offset());
 }
 
+void FloatingObjects::shiftFloatsBy(LayoutUnit blockShift)
+{
+    LayoutUnit shiftX = (m_horizontalWritingMode) ? 0_lu : -blockShift;
+    LayoutUnit shiftY = (m_horizontalWritingMode) ? blockShift : 0_lu;
+
+    for (auto& floater : m_set) {
+        floater->m_frameRect.move(shiftX, shiftY);
```

Comment:
I'd rather use public API (which could do both of these)

Reply:
The reason I didn't go that route was that I didn't want anyone else trying to modify e.g. an individual FloatObject directly, because that could break some invariants in the FloatObjects tracking logic.

Since we're shifting every Float together by the same delta simultaneously, we're not breaking that logic. But if some external caller tried to that, things would likely break. Note that there _are_ setX() and setY() methods on the FloatObject, but I'm specifically bypassing their checks here. I don't want anyone other than this method to be able to do that, though, because this method guarantees they're all moved together.